### DISCUSSION TITLE
VES inversion problem

### DISCUSSION THREAD

**Discussion by TheNablaOperator** [gimli-org/pyGIMLi#567] [Q&A] [answered]
Dear all.
Currently I am having a problem with a VES inversion. 
I have measured data, which should be inverted. However, the inversion is aborted due to "division by zero..."

See the code example below. 
Do you have any ideas? 
Thank you in advance!

``` Python 
import matplotlib.pyplot as plt
import pygimli as pg
import pygimli.meshtools as mt
from pygimli.physics import ert, VESManager
import numpy as np
import math

a = np.linspace(1, 10, 10)
r = [13.652265433675282, 13.758129233697074, 13.206359444328495, 10.380611731594303, 8.330864788419275, 8.317581072952597, 7.120434768503427, 6.0640275619612956, 5.408241121990998, 5.0616707762341955]

for x in range(0,10):
    k = 2*math.pi*a[x]
    rhoa = r[x]*k


test = pg.DataContainer();
test["ab2"] = 3*a
test["mn2"] = a
test["rhoa"] = rhoa
test["k"] = k
test["err"] = ert.estimateError(test, relativeError=0.02, absoluteUError=50e-6)

mgr = VESManager()
model = mgr.invert(test["rhoa"], test["err"], test["ab2"], test["mn2"], nLayers=4, showProgress=1, verbose=1)
```

## PG-Version

1.4.3

## Error

`` ZeroDivisionError: float division by zero ``

**Comment by halbmy** [comment, marked as answer]
First, your rhoa is a float value and it will be overwritten in the loop. I guess you mean
```
k = 2 * np.pi * a
rhoa = r * k
```

If you then use keywords for the geometry,
```
model = mgr.invert(test["rhoa"], test["err"], ab2=test["ab2"], mn2=test["mn2"], nLayers=4, showProgress=1, verbose=1)
```
you will kind of (chi-square of 8) fit your data (`mgr.showFit()`)
![image](https://github.com/gimli-org/gimli/assets/7136401/bdcd8d3e-93dc-4764-a5a3-1d80c6119b34)

with the following model (`mgr.showModel`)
![image](https://github.com/gimli-org/gimli/assets/7136401/311199d3-a231-41be-873b-da9e56a599b0)
